java/arcs/android/storage/database/DatabaseImpl.kt

```diff
@@ -500,17 +509,16 @@ class DatabaseImpl(
                 arrayOf(collectionId.toString())
             )
 
-            // Updated stored version map if different.
-            if (storedVersionMap != data.versionMap) {
-                update(
-                    TABLE_COLLECTIONS,
-                    ContentValues().apply {
-                        put("version_map", data.versionMap.toProtoLiteral())
-                    },
-                    "collection_id = ?",
-                    arrayOf(collectionId.toString())
-                )
-            }
+            // Updated collection metadata.
+            update(
+                TABLE_COLLECTIONS,
+                ContentValues().apply {
+                    put("version_map", data.versionMap.toProtoLiteral())
+                    put("version_number", data.databaseVersion)
+                },
+                "id = ?",
+                arrayOf(collectionId.toString())
+            )
```

Comment:
These are fine for now, but I think In the future we should consider switching to using `rawQuery` and `execSQL` directly for all SQL stuff in the DatabaseImpl (rather than the android-specific update/insert/delete/etc. methods). It would be nice to have total control over the structure of the queries, and to easily be able to copy those queries into a command-line SQLite client to run `EXPLAIN QUERY PLAN` on them.

Reply:
SGTM. I thought the android functions would be better/faster or something, which is mainly why I used them. I can update these in a follow up PR

Reply:
No rush! Just a cleanup task for down the road